rust/main/hyperlane-core/src/chain.rs

```diff
@@ -549,4 +626,32 @@ mod tests {
         );
         assert!("foo".parse::<KnownHyperlaneDomain>().is_err());
     }
+
+    #[test]
+    fn parse_reorg_period() {
+        assert_eq!(
+            serde_json::from_value::<ReorgPeriod>(0.into()).unwrap(),
+            ReorgPeriod::None
```

Comment:
The reason why `0` is parsed to `ReorgPeriod::None` is to maintain compatibility with current configs, where `"reorgPeriod": 0` means no reorg period. Also, `ReorgPeriod::Block(0)` is supposed to be an invalid variant because it’s the same as `ReorgPeriod::None`.

Reply:
+1 would prefer 0 to be ReorgPeriod::None, specifically for the case of lagged calls where we only want to do the whole "get the block number and then subtract the reorg period" bit for non-zero reorg periods.

We could maybe change that logic to be if the reorg period blocks is > 0 instead of `is_none` if you think having `None` as an option is confusing though. But I think having None as a variant can be useful

Reply:
oke sounds good